src/checkpoint/dispatch/dispatch.impl.h

```diff
@@ -50,6 +50,7 @@
 
 #include <stdexcept>
 #include <string>
+#include <optional>
```

Comment:
`std::optional` is documented as being added in C++17, while [Magistrate documentation](https://darma-tasking.github.io/checkpoint_docs/html/index.html) says it's a C++14 library.

Fortunately, it doesn't look like `std::optional` is actually *used* anywhere in here, so this inclusion should just be dropped.

Also, maybe that line of the documentation is out of date, in which case, please open an issue. Additionally, I see two more issues of documentation markup not being rendered properly, which may represent additional issues(s):
* `\checkpoint`
* `@m_class{m-note m-dim}`

Reply:
C++14 thing is a bug, I've opened #371 to track that.

+1 for dropping an unused `#include`.